lib/pbench/test/unit/server/test_datasets_metadata.py

```diff
@@ -401,6 +409,39 @@ def test_put_invalid_deletion(self, query_get_as, query_put_as):
             "user.one": None,
         }
 
+    def test_put_set_errors(self, monkeypatch, query_get_as, query_put_as):
+        real_set = Metadata.setvalue
+
+        @staticmethod
+        def fake_set(dataset: Dataset, key: str, value: Any, user_id: str):
+            if key in ("global.dashboard.seen", "dataset.name"):
+                raise MetadataBadStructure(dataset, key, key)
```

Comment:
Is one of those `key`'s at line 418 supposed to be a `value`?  (Of course, this will affect line 435....)

Reply:
No; in the real exception, we're trying to descend through a JSON hierarchy, and find an intermediate level that's not a `dict`: e.g., I try to set `global.dashboard.seen` when `global.dashboard = 0`. For the fake-out here, I just set out to throw a "plausible" metadata error on one of several key paths, and I arbitrarily used that one with a mock rather than setting up a real failure. (Though if I'd known initially that it was so easy I might not have bothered with the mock at all: at that point I was already mentally committed and it didn't occur to me to back out. 😆 )

That is, I could have just set `global.dashboard.foo = 1` and then set

```json
                "metadata": {
                    "global.dashboard.seen": False,
                    "dataset.name": "foo",
                    "global.dashboard.foo.bar": True,
                }
```

which would have generated a real `MetadataBadStructure(dataset, "global.dashboard.foo.bar", "foo")`